pkg/handler/project/project.go

```diff
@@ -307,3 +308,67 @@ func (h *handler) GetStaffs(c *gin.Context) {
 	c.JSON(http.StatusOK, view.CreateResponse(view.ToProjectMemberListData(members, heads),
 		&view.PaginationResponse{Pagination: query.Pagination, Total: total}, nil, nil))
 }
+
+// DeleteMember godoc
+// @Summary Delete member in a project
+// @Description Delete member in a project
+// @Tags Project
+// @Accept  json
+// @Produce  json
+// @Param Authorization header string true "jwt token"
+// @Param id path string false "member ID"
+// @Success 200 {object} view.DeleteMemberMessageResponse
+// @Failure 400 {object} view.ErrorResponse
+// @Failure 404 {object} view.ErrorResponse
+// @Failure 500 {object} view.ErrorResponse
+// @Router /projects/members/:id [delete]
+func (h *handler) DeleteMember(c *gin.Context) {
+	memberID := c.Param("id")
+
+	// TODO: can we move this to middleware ?
+	l := h.logger.Fields(logger.Fields{
+		"handler":   "project",
+		"method":    "DeleteMember",
+		"projectID": memberID,
+	})
+
+	// get member infor
+	projectMember, err := h.store.ProjectMember.One(h.repo.DB(), memberID)
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			l.Error(err, "project member not found")
+			c.JSON(http.StatusNotFound, view.CreateResponse[any](nil, nil, ErrProjectMemberNotFound, memberID))
+			return
+		}
+		l.Error(err, "error query member from db")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, err, memberID))
+		return
+	}
+
+	if projectMember.Status == model.ProjectMemberStatusInactive {
+		l.Error(ErrInactiveMember, "error inactive member")
+		c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, ErrInactiveMember, memberID))
+		return
+	}
+
+	// Begin transaction
+	tx, done := h.repo.NewTransaction()
+
+	slotID := projectMember.ProjectSlotID
+
```

Comment:
I think the flow is: delete project_member_position -> soft delete project_member -> hard delete project_slot_position -> soft delete project_slot

Reply:
updated

Reply:
@datphamcode295 should confirm flow with @namnhce before implement it

Reply:
missing delete project head

Reply:
updated